### DISCUSSION TITLE
v1.114.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#12393] [Announcements]
# v1.114.0

## Highlights

Welcome to release `v1.114.0` of Immich. Who doesn't love a juicy release on a Friday? This release focuses on additional enhancements and bug fixes for recent new features. Please find some of the highlights below:

- Tag improvements (including Lightroom support)
- Import faces from EXIF and XMP sidecars (Digikam format)
- Better handling of timezones
- Upload panel \*\*New look\*\*
- Automatic database reconnection

### Tag improvements

Since the last release, when we introduced support for tags, we have made a few changes based on your feedback.

- We now support reading from `HierarchicalSubject`. Values will be parsed as hierarchical tags, with `|` as a delimiter.
- We now only read from `Keywords` as a fallback when no hierarchical tag information is available
- We have fixed a bug where you could not remove the last tag from EXIF/XMP.

NOTE: these changes apply to the metadata extraction job. To apply them to your photos and videos, re-run metadata extraction.

### Import faces from metadata

This release adds the ability to read face metadata in Digikam format during metadata extraction, increasing interoperability with other software and laying the foundation for faces from sources other than machine learning.

Note that there are certain limitations with these faces. These limitations may be removed in a later release:
- Assets that contain face metadata are skipped during face detection
- Faces from metadata are not considered during facial recognition
- Faces from metadata are grouped into people based on name
- Only face metadata in Digikam format is supported

You can enable this feature by going to `Administration > Metadata Settings`.

### Better handling of timezones

This version fixes some issues related to how timezone offsets are parsed and how timezones and offsets are displayed. The offset is now parsed correctly for photos and videos taken with the "+00:00" offset (e.g., several African countries, Iceland, Portugal/UK/Ireland in winter).

If you notice that the "+00:00" offset is not shown correctly, please re-run the "Extract Metadata" job on all photos and videos.

### Upload panel
The upload panel has a new look! It works the same as before, except that duplicates are no longer automatically cleared from the queue. They will remain behind, along with a clickable icon, that links to the deduped image.

| Light | Dark |
| - | - |
| ![image](https://github.com/user-attachments/assets/2911ed68-6ea2-4688-9b6e-57df7b801fc8) | ![image](https://github.com/user-attachments/assets/9f7a65f2-6f4a-4112-bc0c-a0f43ca83716) |

### Automatic database reconnection
Previously, when the connection to the database was interrupted for any reason (restart, timeout, network interruption, etc.), the server would never reconnect. Restoring the connection required a container restart. Now, the server will automatically try to reconnect every 5 seconds indefinitely.

![image](https://github.com/user-attachments/assets/c0814c51-a4b0-45ac-8875-4c837acdedf9)

- - - -

## Support Immich

<p align="center">
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich by purchasing a product key at [https://buy.immich.app](https://buy.immich.app).

Cheers! 🍻

### 🚀 Features
* feat(server): Import face regions from metadata by @bugfest in https://github.com/immich-app/immich/pull/6455
### 🌟 Enhancements
* feat(server): support lightroom tags by @jrasm91 in https://github.com/immich-app/immich/pull/12288
* feat(web): show folder navigation in root directory by @ben-basten in https://github.com/immich-app/immich/pull/12299
* chore(mobile): handle sync album on duplicated by @alextran1502 in https://github.com/immich-app/immich/pull/12173
* feat(web): sort albums in modal by @jrasm91 in https://github.com/immich-app/immich/pull/12331
* feat(web): add skip link to sidebar by @ben-basten in https://github.com/immich-app/immich/pull/12330
* fix(server): search suggestions include partner assets by @Lukasdotcom in https://github.com/immich-app/immich/pull/12269
* feat(web): add download shortcut on the timeline & asset viewer by @imondrag in https://github.com/immich-app/immich/pull/12339
### 🐛 Bug fixes
* fix(server): empty trash for archived assets by @alextran1502 in https://github.com/immich-app/immich/pull/12281
* feat(server): prefer tagslist by @jrasm91 in https://github.com/immich-app/immich/pull/12286
* fix: remove public. reference in migration sql by @alextran1502 in https://github.com/immich-app/immich/pull/12285
* fix(server): public references in migrations by @jrasm91 in https://github.com/immich-app/immich/pull/12298
* fix(web): paste event in input fields by @jrasm91 in https://github.com/immich-app/immich/pull/12297
* fix(server): parse time zone with explicit zero offset by @C-Otto in https://github.com/immich-app/immich/pull/12307
* fix(web): retain selected time zone offset also for +00:00 by @C-Otto in https://github.com/immich-app/immich/pull/12310
* fix(web): handle RTL languages in the map component by @zackpollard in https://github.com/immich-app/immich/pull/12308
* fix(server): auto-reconnect to database by @jrasm91 in https://github.com/immich-app/immich/pull/12320
* fix(web): correct color for active tree item by @alextran1502 in https://github.com/immich-app/immich/pull/12318
* refactor(web): upload panel by @jrasm91 in https://github.com/immich-app/immich/pull/12326
* fix(mobile): background task crashing on Android by @alextran1502 in https://github.com/immich-app/immich/pull/12314
* fix(server): asset no longer has tags by @jrasm91 in https://github.com/immich-app/immich/pull/12350
* fix(web): only show valid time zones/offsets, update list based on date by @C-Otto in https://github.com/immich-app/immich/pull/12315
* fix: flash bug on tag by @midzelis in https://github.com/immich-app/immich/pull/12332
* fix(web): showing album timeline after adding new assets by @alextran1502 in https://github.com/immich-app/immich/pull/12354
* fix(mobile): download asset to DCIM/Immich folder on Android by @alextran1502 in https://github.com/immich-app/immich/pull/12355
* fix(web): person asset grid by @jrasm91 in https://github.com/immich-app/immich/pull/12370
* fix(server): copy video projection metadata for 360 videos by @mertalev in https://github.com/immich-app/immich/pull/12376
* fix(web): auth on navigation from shared link to timeline by @michelheusschen in https://github.com/immich-app/immich/pull/12385
* fix(web): broken album thumbnail by @michelheusschen in https://github.com/immich-app/immich/pull/12381
* fix(web): ensure shared link covers are full size by @michelheusschen in https://github.com/immich-app/immich/pull/12386
* fix(web): show focus outline for asset thumbnails again by @michelheusschen in https://github.com/immich-app/immich/pull/12382
### 📚 Documentation
* docs: Fix link to ioredis docs by @halkeye in https://github.com/immich-app/immich/pull/12291
* docs: Add conditional album storage template information by @feyst in https://github.com/immich-app/immich/pull/12218
### 🌐 Translations
* feat(web): add Malay language by @jrasm91 in https://github.com/immich-app/immich/pull/12311
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/12265

## New Contributors
* @halkeye made their first contribution in https://github.com/immich-app/immich/pull/12291
* @bugfest made their first contribution in https://github.com/immich-app/immich/pull/6455
* @Lukasdotcom made their first contribution in https://github.com/immich-app/immich/pull/12269

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.113.1...v1.114.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.114.0'>v1.114.0</a>.</em>

**Comment by maxshcherbina** [comment]
Thanks Immich team for the quick turnaround on bugs and features! <3

**Comment by ok1ok1** [comment]
Nice update. Just curious, is it possible to search by embedded keywords? It seems they get picked up as tags but so far as I can tell they are not searchable. Maybe I'm missing something.

**Reply by mertalev** [reply]
Not yet, but this is definitely planned.

**Comment by aamirazad** [comment]
> Who doesn't love a juicy release on a Friday? 

Nothing has gone wrong by deploying on a friday..

**Comment by chriswa81** [comment]
Would it be possible in future not only to regularly check the connection to the database and re-connect if necessary, but also the connection to the mounted external upload target? 
If the connection is interrupted, I can mount it again using a cronjob. But I still have to restart the immich container once afterwards. 
Would a reconnection also be possible here in the future?

Thank you very much.

**Comment by DeltaTango69** [comment]
In Google Play Store no update is available. It's still the 1.113.1?

**Comment by sigulete** [comment]
The link to Repair page has disappeared.
I still can access repair `https://<my-server>/admin/repair` but there is no item in the menu.

**Reply by mio-19** [reply]
https://github.com/immich-app/immich/pull/12294

**Comment by mio-19** [comment]
Full Changelog link is broken. It links to `https://github.com/immich-app/immich/compare/v1.113.1`. The correct link might be `https://github.com/immich-app/immich/compare/v1.113.1...main`

**Reply by jrasm91** [reply]
Fixed

**Reply by mio-19** [reply]
The changelog link still renders incorrectly on my browser 
<img width="1099" alt="Screenshot 2024-09-11 at 2 42 47 PM" src="https://github.com/user-attachments/assets/56f2ae1c-8f51-47f9-848d-ccf996386064">

It still links to https://github.com/immich-app/immich/compare/v1.113.1

**Reply by jrasm91** [reply]
Fixedddddddd

**Comment by rbrianza** [comment]
Thank you for this awesom app, i am in the progress of moving from google photos to immich.
I have some questions concerning the tags functionality:
1. Is there a possibility to add selected Tags to a shared Album?
The popup menu for the selected items is missing on `https://<my-server>/tags?path=<tagname>`
2. Is it possible to also view the tagged photos from the shared partner?
On `https://<my-server>/tags?path=<tagname>` i only see mine
3. Is it possible to also add tags to shared photos from a partner?

Thanks in advance for your reply

**Reply by C-Otto** [reply]
Short answer: No. Longer answer: for shared assets (or albums), you only get to see the asset itself. You don't share any of the other features (tags, faces, rating, ...). I, personally, would also like to see this change, which might happen in a future version.

**Comment by dlang123** [comment]
Tags, I had previously done an import of Metadata and have lots of tags for Instance, on pics that were Studio,PinkDress,5x7 in lightroom those were hierarchy tags. We would get a Tree that was 
Studio
-PinkDress
--5x7

But the original import made them one Tag, and then with the Tag fixes now they are individual tags. 2 Questions here:
1) Do I have to manually delete the single tags now? the Studio,PinkDress,5x7 one because now I have a Studio tag, a PinkDress tag, and a 5x7 tag.
2) Any thoughts on having a tag hierarchy like Lightroom use to do?

**Reply by C-Otto** [reply]
Re 1: I think so, yeah. If you didn't do a lot of manual work, you could delete all tags and start over. The new code with the fixes should not re-create the old tags.

Re 2: https://github.com/immich-app/immich/pull/12288

**Reply by dlang123** [reply]
I guess the better question on #2 is web navigation. Any plans/thoughts to have a tree instead of a flat list?

**Reply by mertalev** [reply]
It should use a tree view for hierarchical tags. Are you saying it thinks those are three independent tags? That sounds like a bug if so.

**Reply by C-Otto** [reply]
@jrasm91 that's your turf, care to chime in?

**Reply by jrasm91** [reply]
In the near future (TM) we'll add a nightly job that will delete empty tags, which should take care of (1). For (2) we support nested tags, even from lightroom, so I'm not sure what are you asking for exactly. You can see the structure here - https://my.immich.app/tags

**Reply by jrasm91** [reply]
Tags are extracted during metadata extraction so you obviously have to re-run those jobs after upgrading to the release with the changes we're talking about.

**Reply by dlang123** [reply]
> In the near future (TM) we'll add a nightly job that will delete empty tags, which should take care of (1). For (2) we support nested tags, even from lightroom, so I'm not sure what are you asking for exactly. You can see the structure here - https://my.immich.app/tags

That link doesn't work for me.

Here is what I see.  In the left 
![image](https://github.com/user-attachments/assets/a413e1eb-36e9-4781-8e74-023745c9b4af)

You see the tag drone tag has pictures, and the old tag from before the hierarchy was fixed was drone,neighborhood which has no pictures now since I reran metadata extraction. No problem I will delete it.

But I have a neighborhood tag that has the subset of drone, neighborhood pics that were there originally.
![image](https://github.com/user-attachments/assets/a783dbfe-1b4a-4efe-95c5-1c243d37af91)

So basically I have an alphabetical list on the browser side that has bread crumb navigation.  This could just be a me thing, but it sure would be nice if they were associated in the view like this

![image](https://github.com/user-attachments/assets/ed5f2d87-8dd0-432a-836f-6805d37ad606)

**Reply by dlang123** [reply]
Adding to it.. I dropped one of the images into exiftool to see what its exif data is and it has the following tags:
Subject                         : drone, neighborhood
Hierarchical Subject            : drone, neighborhood
Keywords                        : drone, neighborhood

**Reply by jrasm91** [reply]
Sounds like you need to rerun metadata extraction.

**Reply by dlang123** [reply]
> Sounds like you need to rerun metadata extraction.

I'll run it again. I ran it once after I upgraded to 1.115, but will try again.

**Reply by jrasm91** [reply]
Maybe you can upload a test image here as a zip and I can take a look. 

Also you can rerun metadata extraction for one asset from the /photos page, but if you did it in 115 already something else is likely going on.

**Reply by dlang123** [reply]
[drone sunset (23 of 26).zip](https://github.com/user-attachments/files/17004176/drone.sunset.23.of.26.zip)
Yeah I have run it a few times. We could just be talking past each other or maybe I don't understand how the tags show for you guys.

**Reply by jrasm91** [reply]
![image](https://github.com/user-attachments/assets/6d2419e7-3b01-4e9b-9e87-25a069e24226)

**Comment by jrasm91** [comment]
None of these of nested tags:

```
  "Subject": ["dji","drone","sunset"],
  "HierarchicalSubject": ["dji","drone","sunset"],
  "Keywords": ["dji","drone","sunset"],
```

**Reply by dlang123** [reply]
> None of these of nested tags:
> 
> ```
>   "Subject": ["dji","drone","sunset"],
>   "HierarchicalSubject": ["dji","drone","sunset"],
>   "Keywords": ["dji","drone","sunset"],
> ```

Thanks. That explains it. They were at one time, something must of changed them before Immich.
